lib/http/response.rb

```diff
@@ -96,8 +96,13 @@ def flush
     #   (not an integer, e.g. empty string or string with non-digits).
     # @return [Integer] otherwise
     def content_length
+      # http://greenbytes.de/tech/webdav/rfc7230.html#rfc.section.3.3.3
+      # Clause 3: "If a message is received with both a Transfer-Encoding
+      # and a Content-Length header field, the Transfer-Encoding overrides the Content-Length.        
+      return nil if @headers.include?(Headers::TRANSFER_ENCODING) ||
+                    !@headers.include?(Headers::CONTENT_LENGTH)
```

Comment:
The only thing is that if that is gonna happen (change to underlying data structure) it will be released only as next major release due to introduction of backward incompatible changes.